File: 数组/双指针/11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans=-999;
        int left, right;
        left = 0;
        right = height.size()-1;

        while (left < right){
            int size = (right-left) * (min(height[left], height[right]));
            ans = max(ans, size);

            if (height[left] > height[right]){
                right--;
            }
            else {
                left++;
            }
        }

        return ans;
    }
};

File: 数组/双指针/11.java
class Solution {
    public int maxArea(int[] height) {
        int ans=-999999;
        int l=0, r=height.length-1;
        while (l < r) {
            int now = Math.min(height[l], height[r]) * (r - l);
            if (now > ans) {
                ans = now;
            }
            if (height[l] < height[r]) {
                l++;
            } else {
                r--;
            }
        }
        return ans;
    }
}
